whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java

```diff
@@ -0,0 +1,35 @@
+package net.ripe.db.whois.common.dao.jdbc.index;
+
+import net.ripe.db.whois.common.dao.RpslObjectInfo;
+import net.ripe.db.whois.common.rpsl.AttributeParser;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import net.ripe.db.whois.common.rpsl.attrs.AttributeParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+class IndexWithEmailAddressValueAndType extends IndexWithValueAndType {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IndexWithEmailAddressValueAndType.class);
+
+    private static final AttributeParser.EmailParser EMAIL_PARSER = new AttributeParser.EmailParser();
+
+    public IndexWithEmailAddressValueAndType(final AttributeType attributeType, final String lookupTableName, final String lookupColumnName) {
+        super(attributeType, lookupTableName, lookupColumnName);
+    }
+
+    @Override
+    public int addToIndex(JdbcTemplate jdbcTemplate, RpslObjectInfo objectInfo, RpslObject object, String value) {
```

Comment:
shoudln't we make sure that this function is called for email type only?

Reply:
That's done in `IndexStrategies.java` you define which strategy to use for different attribute types. This class is only added to attributes that contain an email address.

Reply:
Normalise email address function can be static to make sure both the functions are doing same thing and code is not duplicated. May be shift it to util class?

Reply:
Good idea. I'll make it static but I'll leave it in one of the Index classes and call it from the other one

Reply:
Updated code changes